### PR TITLE
[UX] Add home button

### LINKED ISSUE #3200: anyway to shortcut to home directory?
* KOReader version: last
* Device: all

#### Issue
It's always annoying me when back from sub directory to home.
Do it have a shortcut in program that I don't know?

I think empty space on to of file browser is best place for this.
But anywhere is fine.

### PR DESCRIPTION
**Pull request by Frenzie**
Tap to go HOME. Hold to set current folder as HOME. When no HOME is set yet tap also asks to set current folder as HOME.

See https://github.com/koreader/koreader/issues/2957#issuecomment-308513062

Closes #3200.

screenshot_2017-09-26_18-14-30

### CODE REVIEW

**Comment by Frenzie** [comment]
I'm not 100% happy with how it looks; suggestions welcome.

Also I forgot to include the obligatory before screenshot.

screenshot_2017-09-26_18-27-39

**Comment by Frenzie** [comment]
True, maybe it's just that. It occurs to me that our former exit icon would fit the job perfectly.

appbar home

https://github.com/koreader/koreader/blob/62754538557437ead656b08874ef247cff4d1e69/resources/icons/appbar.home.png

**Comment by Frenzie** [comment]
Alright, that's better.

screenshot_2017-09-26_18-58-04

**Comment by Frenzie** [comment]
I think something like this is a pretty good look.

screenshot_2017-09-27_00-14-15
screenshot_2017-09-27_00-14-25
screenshot_2017-09-27_00-14-39

Also I included a prototype for better padding handling which I'd like to stick in some key widgets (Button, FrameContainer). Could you have a look at the basic idea, @poire-z @robert00s ? The super complex layouts with all kinds of manual Horizontal/VerticalSpan insertions are horribly overcomplicated when you could just stick a padding_left=0 on the first element and a padding_right=0 on the last.

File: frontend/ui/widget/iconbutton.lua [hunk 1]
```diff
@@ -31,25 +38,42 @@ function IconButton:init()
...
+    table.insert(self.button, VerticalSpan:new{})
 
     self[1] = self.button
     self:update()
 end
 
 function IconButton:update()
-    self.button[1].width = self.horizontal_padding
-    self.button[3].width = self.horizontal_padding
+    if self.horizontal_padding > 0 then self:setHorizontalPadding(self.horizontal_padding) end
```
Comment (poire-z) [inline frontend/ui/widget/iconbutton.lua]: Looks fine. May be you could just get rid of self.horizontal_padding, and use padding_left and padding_right in touchmenu.

Comment (Frenzie) [reply frontend/ui/widget/iconbutton.lua]: Yes, I just wanted this to be mergeable.

Comment (Frenzie) [reply frontend/ui/widget/iconbutton.lua]: @poire-z Is there a particular reason you didn't just do this?

```lua
            menu_items[c].padding_left = padding
            menu_items[c].padding_right = padding
            -- or rather horizontal_padding
            menu_items[c]:update()
```

Comment (poire-z) [reply frontend/ui/widget/iconbutton.lua]: Is this still a question ? (hard to follow your thoughts with the multiple crosslinks :)
If yes, well, I didnt because they weren't there cause you just added those padding_left & co :)
And before that, I would have had to do instead:
```
menu_items[c].button[1].width = padding
menu_items[c].button[3].width = padding
menu_items[c]:update()
```
which expose too much of the internals.
Or I misunderstood your question ?

Comment (Frenzie) [reply frontend/ui/widget/iconbutton.lua]: I was lazy and copied my new code. I just meant

menu_items[c].horizontal_padding = padding
menu_items[c]:update()

Anyway, if you have no objections I reckon this is ready to be merged?

Comment (poire-z) [reply frontend/ui/widget/iconbutton.lua]: OK. My thinking was just to not expose the existence of the internal update(), and provide some public method setHorizontalPadding() for that.
But in your case, as padding_left & co are known, I guess it's ok for update() to be known and called. So no objection to merge.
